types/webpack/index.d.ts

```diff
@@ -1297,7 +1297,7 @@ declare namespace webpack {
             prefetched: boolean;
             profile: any; // TODO
             providedExports?: any; // TODO
-            reasons: {
+            reasons: [{
```

Comment:
if it can be an object like this or an array of objects like this then you need to put this type into a type reference or interface and then use it here. Something like:

```ts
type Reason = {
    explanation: string | undefined;
    // ...
};

```
```suggestion
            reasons: Reason[] | Reason
```

Reply:
Seems it can't be an object and always is an array. At least on [webpack](https://github.com/webpack/webpack/blob/master/lib/Stats.js#L535) current version. 

Extract `Reason` type is a great idea.